cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -1133,10 +1133,11 @@ buildInplaceUnpackedPackage verbosity
       | elabBuildHaddocks pkg = action
       | otherwise            = return ()
 
-    whenReRegister  action = case buildStatus of
-      BuildStatusConfigure          _ -> action
-      BuildStatusBuild Nothing      _ -> action
-      BuildStatusBuild (Just mipkg) _ -> return mipkg
+    whenReRegister  action
+      = case buildStatus of
+          BuildStatusBuild (Just mipkg) _ -> return mipkg
+          _ | null (elabBuildTargets pkg) -> return Nothing -- ick
```

Comment:
You're right, it is no longer used, so it's not necessary to collect and return the `InstalledPackageInfo`.